cmd/package_managers_test.go

```diff
@@ -161,6 +162,144 @@ func Test_fetchGitRepositoryFromNPM(t *testing.T) {
 	}
 }
 
+func Test_findGitRepositoryInPYPIResponse(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name                string
+		partialPYPIResponse string
+		want                string
+		wantErrStr          string
+	}{
+		{
+			name: "findGitRepositoryInPYPIResponse_none",
+			partialPYPIResponse: `
+ {
+  "info": {
+    "platform": "UNKNOWN",
+    "not_a_project_url": "https://github.com/htaslan/color",
+    "project_urls": {
+      "Homepage": "http://git_NOT_VALID_hub.com/htaslan/color"
+    }
+  }
+}
+`,
+			want:       "",
+			wantErrStr: "could not find source repo for pypi package: somePackage",
+		},
+		{
+			name: "findGitRepositoryInPYPIResponse_project_url",
+			partialPYPIResponse: `
+ {
+  "info": {
+    "platform": "UNKNOWN",
+    "project_url": "https://github.com/htaslan/color/",
+    "project_urls": {
+      "Homepage": "http://git_NOT_VALID_hub.com/htaslan/color"
+    }
+  }
+}
+`,
+			want:       "https://github.com/htaslan/color",
+			wantErrStr: "",
+		},
+		{
+			name: "findGitRepositoryInPYPIResponse_project_urls",
+			partialPYPIResponse: `
+ {
+  "info": {
+    "platform": "UNKNOWN",
+
+    "project_url": "http://git_NOT_VALID_hub.com/htaslan/color",
+    "project_urls": {
+      "RandomKey": "https://github.com/htaslan/color/",
+      "SponsorsIgnored": "https://github.com/sponsors/htaslan",
+      "AnotherRandomKey": "http://git_NOT_VALID_hub.com/htaslan/color"
+    }
+  }
+}
+`,
+			want:       "https://github.com/htaslan/color",
+			wantErrStr: "",
+		},
+		{
+			name: "findGitRepositoryInPYPIResponse_dedup",
+			partialPYPIResponse: `
+ {
+  "info": {
+    "platform": "UNKNOWN",
+    "project_url": "foo",
+    "project_urls": {
+      "RandomKey": "https://github.com/htaslan/color/",
+      "AnotherRandomKey": "http://htaslan.github.io/color",
+      "CapsTestKey": "http://HTASLAN.github.io/cOLOr",
+      "TrailingGit": "https://github.com/htaslan/color.git"
+    }
+  }
+}
+`,
+			want:       "https://github.com/htaslan/color",
+			wantErrStr: "",
+		},
+		{
+			name: "findGitRepositoryInPYPIResponse_dedup_gitlab",
+			partialPYPIResponse: `
+ {
+  "info": {
+    "platform": "UNKNOWN",
+    "project_url": "foo",
+    "project_urls": {
+      "RandomKey": "https://gitlab.com/htaslan/color/",
+      "raNdoMkEY": "https://gitlab.com/hTASLan/color"
+    }
+  }
+}
+`,
+			want:       "https://github.com/htaslan/color",
+			wantErrStr: "",
```

Comment:
apologies for all the back and forth, the CI requiring approval for first-time contributors sometimes lets copy/paste bugs fall through the cracks.

Reply:
Sigh. No that was my fault. confirm from cmd/ 

go test
go fmt
gofumpt -d package* 

all succeed